src/main/java/org/traccar/protocol/GranitProtocolDecoder.java

```diff
@@ -209,7 +209,7 @@ protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg
                     if (buf.getUnsignedByte(buf.readerIndex()) != 0xFE) {
                         Position position = new Position(getProtocolName());
                         position.setDeviceId(deviceSession.getDeviceId());
-                        position.setTime(new Date((unixTime + i * timeIncrement) * 1000));
+                        position.setTime(new Date((unixTime + (long) i * timeIncrement) * 1000));
```

Comment:
Why do we need this?

Reply:
`i `and `timeIncrement` are both integers, their multiplication can overflow.

Reply:
ok, shall I revert this line and the rest of the PR look good to you?